tests/integration_tests/instances.py

```diff
@@ -201,7 +201,9 @@ def ip(self) -> str:
         if self._ip:
             return self._ip
         try:
-            self._ip = self.instance.ip
+            # in some cases that ssh is not used, an address is not assigned
+            if self.instance.execute_via_ssh:
```

Comment:
I think `execute_via_ssh` is only defined for LXD instances, so as-is I think this line would raise an `AttributeError` on other clouds.

Reply:
Thanks! I think I fixed it.